### PR TITLE
Speaker Service

### PR DESCRIPTION
**Pull request by AndyGee**
JavaOne speaker page parser
Arquillian REST Test
File normalization

### CODE REVIEW

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/ProducerVenue.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import java.net.MalformedURLException;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+
+public class ProducerVenue {
+
+    private final Logger log = LogManager.getLogManager().getLogger(ProducerVenue.class.getName());
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/ProducerVenue.java]: Should be static?

File: microservice-speaker/microservice-speaker-web/pom.xml [hunk 2]
```diff
@@ -59,11 +59,38 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.el</groupId>
+            <artifactId>el-api</artifactId>
+            <version>2.2</version>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish.web</groupId>
+            <artifactId>el-impl</artifactId>
+            <version>2.2</version>
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/pom.xml]: I assume this is the RI for the EL spec?

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/QVenue.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Qualifier
+@Retention(RUNTIME)
+@Target({ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER, ElementType.TYPE})
+public @interface QVenue {
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/QVenue.java]: What's a QVenue?

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class VenueJavaOne2016 extends Venue {
+
+    private static final String JAVA_ONE_2016 = "JavaOne2016";
+    private final Logger log = Logger.getLogger(VenueJavaOne2016.class.getName());
+
+    private final URL url = URI.create("https://www.oracle.com/javaone/speakers.html").toURL();
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java]: static?

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+    VenueJavaOne2016() throws MalformedURLException {
+    }
+
+    public static void main(final String[] args) {
+        try {
+            new VenueJavaOne2016().getSpeakers();
+        } catch (final MalformedURLException e) {
+            e.printStackTrace();
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java]: Logging?

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        return JAVA_ONE_2016;
+    }
+
+    public Set<Speaker> getSpeakers() {
+
+        final Set<Speaker> speakers = new HashSet<>();
+
+        InputStream is = null;
+
+        try {
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java]: Could use TWR

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/persistence/SpeakerDAO.java [hunk 7]
```diff
@@ -15,47 +15,74 @@
...
-        s.setPicture("https://www.oracle.com/us/assets/cw16-sebastian-daschner-3102339.png");
-        return s;
+    public void remove(final String id) {
+
     }
 
-    public Optional<Set<Speaker>> getSpeakers(final String venue) {
+    public Speaker update(final Speaker speaker) {
+
+        final Iterator<Speaker> iterator = this.speakers.get(this.venue.getName()).iterator();
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/persistence/SpeakerDAO.java]: If we're dealing with Iterators, can we not move to Streams?

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/rest/Resource.java [hunk 8]
```diff
@@ -19,32 +19,71 @@
...
 
 /**
  * The Speaker resource
  */
 @Singleton
+@Startup
 @Lock(LockType.READ)
+@Produces(MediaType.APPLICATION_JSON)
 @Path("/")
 public class Resource {
```
Comment (karianna) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/rest/Resource.java]: Is Resource the right name for this class?  We're dealing with processing Speaker related requests

**Comment by karianna** [comment]
Weird, the individual comments I added aren't coming through in the GitHub UI.  Anyhow, feel free to ignore if you wish :-)

File: microservice-speaker/microservice-speaker-web/pom.xml (hunk 2 above)
Comment (AndyGee) [inline microservice-speaker/microservice-speaker-web/pom.xml]: Yep, we have the tomee:javaee-api shaded with it in, will update it next iteration

File: microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java (hunk 5 above)
Comment (AndyGee) [inline microservice-speaker/microservice-speaker-web/src/main/java/io/microprofile/showcase/speaker/domain/VenueJavaOne2016.java]: Only in the main, it's never called outside

**Pull request merged**